src/LibraryViewExtensionMSWebBrowser/LibraryViewController.cs

```diff
@@ -0,0 +1,581 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Security.Permissions;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Threading;
+using Dynamo.Extensions;
+using Dynamo.LibraryViewExtensionMSWebBrowser.Handlers;
+using Dynamo.LibraryViewExtensionMSWebBrowser.ViewModels;
+using Dynamo.LibraryViewExtensionMSWebBrowser.Views;
+using Dynamo.Logging;
+using Dynamo.Models;
+using Dynamo.Search;
+using Dynamo.Search.SearchElements;
+using Dynamo.ViewModels;
+using Dynamo.Wpf.Interfaces;
+using Dynamo.Wpf.ViewModels;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Dynamo.LibraryViewExtensionMSWebBrowser
+{
+    /// <summary>
+    /// This object is exposed to the browser to pass data back and forth between contexts.
+    /// </summary>
+    [PermissionSet(SecurityAction.Demand, Name = "FullTrust")]
+    [ComVisibleAttribute(true)]
+    public class scriptingObject
+    {
+        private LibraryViewController controller;
+
+        public scriptingObject(LibraryViewController controller)
+        {
+            this.controller = controller;
+        }
+
+        /// <summary>
+        /// Main callback which libraryJS uses to notify the extension to perform some action.
+        /// </summary>
+        /// <param name="data"></param>
+        public void notify(string data)
+        {
+            scriptNotifyHandler(data);
+        }
+
+        /// <summary>
+        /// Used to get access the iconResourceProvider and return a base64encoded string version of an icon.
+        /// </summary>
+        /// <param name="iconurl"></param>
+        public void GetBase64StringFromPath (string iconurl)
+        {
+            
+            string ext;
+            var iconAsBase64 = controller.iconProvider.GetResourceAsString(iconurl, out ext);
+            if (string.IsNullOrEmpty(iconAsBase64))
+            {
+                controller.browser.InvokeScript("completeReplaceImages", "");
+            }
+            if (ext.Contains("svg"))
+            {
+                ext = "svg+xml";
+            }
+            //send back result.
+            controller.browser.InvokeScript("completeReplaceImages", $"data:image/{ext};base64, {iconAsBase64}");
+        }
+
+        private void scriptNotifyHandler(string dataFromjs)
+        {
+
+            if (dataFromjs == "requestUpdateLibrary")
+            {
+                controller.RefreshLibraryView(controller.browser);
+                
+            }
+            //a more complex action needs to be taken on the c# side.
+            else if (!string.IsNullOrEmpty(dataFromjs))
```

Comment:
You could invert the `if` condition by simply returning from the method for a null or empty string?